documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,452 @@
+If we click on the `Dev Team A` policy, we see that it matches all users that have a realm role called `Dev Team A`.
+
+Similarly, the `Dev Team B ...` permissions perform matching using the `Dev Team B` policy which also uses realm role to match allowed users - in this case those with realm role `Dev Team B`.
+The `Dev Team B ...` permissions grant users `Describe` and `Read` on `Topic:x_*`, and `Group:x_*` resources, effectively giving matching users and clients the ability to read from topics, and update the consumed offsets for topics and consumer groups that have names starting with 'x_'.
+
+## Targeting Permissions - Clients and Roles vs. Users and Groups
+
+In Keycloak, confidential clients with 'service accounts' enabled can authenticate to the server in their own name using a clientId and a secret.
+This is convenient for microservices which typically act in their own name, and not as agents of a particular user (like a web site would, for example).
+Service accounts can have roles assigned like regular users.
+They cannot, however, have groups assigned.
+As a consequence, if you want to target permissions to microservices using service accounts, you cannot use Group policies, and should instead use Role policies.
+Or, thinking about it another way, if you want to limit certain permissions only to regular user accounts where authentication with username and password is required, you can achieve that as a side effect of using the Group policies, rather than the Role policies.
+That's what we see used in `permissions` that start with 'ClusterManager'.
+Performing cluster management is usually done interactively - in person - using CLI tools.
+It makes sense to require the user to log-in, before using the resulting access token to authenticate to the Kafka Broker.
+In this case the access token represents the specific user, rather than the client application.
+
+
+## Authorization in Action Using CLI Clients
+
+A note of caution - this example uses the latest version of Keycloak, and relies on bugfixes of issues in some older versions.
+Let's make sure the Authorization rules have been properly imported when the Keycloak was started.
+
+Under `Clients` / `kafka` / `Authorization` / `Settings` make sure the `Decision Strategy` is set to `Affirmative`, and NOT to `Unanimous`.
+Click on other tabs and make sure there are some resources, authorization claims, policies and permissions defined.
+
+With configuration now in place, let's create some topics, use a producer, a consumer, and try to perform some management operations using different user and service accounts.
+
+First, we run a new interactive pod container using a Strimzi Kafka image which we use to connect to the already running Kafka broker.
+
+    kubectl run -ti --rm --restart=Never --image=quay.io/strimzi/kafka:latest-kafka-2.6.0 kafka-cli -- /bin/sh
```

Comment:
```suggestion
First, a new interactive pod container is run using a Strimzi Kafka image to connect to a running Kafka broker.

[source,shell]
----
kubectl run -ti --rm --restart=Never --image=quay.io/strimzi/kafka:latest-kafka-2.6.0 kafka-cli -- /bin/sh
----
```

Reply:
Would this section be okay for a downstream doc? Are we expecting people to step through the example or is it intended for reference? Impact on testing.